frontend/src/lib/utils.tsx

```diff
@@ -1191,19 +1191,25 @@ export function toString(input?: any | null): string {
     return input?.toString() || ''
 }
 
+export function skipNaNs(input: any[]): number[] {
+    return input.filter((value) => !isNaN(value))
+}
+
+/**
+ * Returns the average of an array
+ * @param input e.g. [100,50, 75]
+ */
 export function average(input: number[]): number {
-    /**
-     * Returns the average of an array
-     * @param input e.g. [100,50, 75]
-     */
+    input = skipNaNs(input)
     return Math.round((input.reduce((acc, val) => acc + val, 0) / input.length) * 10) / 10
```

Comment:
are we changing the result correctly here....

In the description it says

> skip NaNs ... and fall back only if the value is really unknown (not just 0)

Equating NaN with zero

But we're changing the array length too. And zeroes contribute to the average.

I'm not clear on where the NaNs are coming from but, if they are equivalent to zero, we should be checking original array length for the average?